tests/core/conftest.py

```diff
@@ -269,3 +347,55 @@ def protect_response(r):
     if r.status_code >= 300:
         message = 'Server responded with {r.status_code}\nbody:\n{r.text}'
         raise ValueError(message)
+
+
+def cleanup_dind(admin_mc, cluster, cluster_file):
+    admin_mc.client.delete(cluster)
+    remove_cluster(cluster_file)
+
+
+def create_kubeconfig(dind_cc, headers):
+    # wait for permissions to sync
+    time.sleep(1)
+
+    # request cluster scoped kubeconfig
+    KUBE_URL = BASE_URL + \
+        '/clusters/' + \
+        dind_cc.cluster.id + \
+        '?action=generateKubeconfig'
+    r = requests.post(KUBE_URL, headers=headers, verify=False)
+    protect_response(r)
+    cluster_kubeconfig = r.json()['config']
+
+    # write cluster scoped kubeconfig
+    cluster_kubeconfig_file = "kubeconfig-" + \
+        str(random.randint(10000, 99999)) + ".yml"
+    f = open(cluster_kubeconfig_file, "w")
+    f.write(cluster_kubeconfig)
+    f.close()
+    config = yaml.safe_load(cluster_kubeconfig)
+    token_name = config['users'][0]['user']['token'].split(':')[0]
+
+    # wait for token to sync
+    crd_client = CustomObjectsApi(
+        kubernetes_api_client(
+            dind_cc.admin_mc.client,
+            dind_cc.cluster.id
+        )
+    )
+
+    def cluster_token_available():
+        try:
+            return crd_client.get_namespaced_custom_object(
+                'cluster.cattle.io',
+                'v3',
+                'cattle-system',
+                'clusterauthtokens',
+                token_name
+            )
+        except Exception:
```

Comment:
This should catch `ApiException`, naked `Exception` will mask anything else that could throw.